modules/azure/src/main/java/org/testcontainers/containers/AzuriteContainer.java

```diff
@@ -0,0 +1,529 @@
+
+    static final String DEFAULT_HOST = "127.0.0.1";
+
+    static final int DEFAULT_BLOB_PORT = 10000;
+
+    static final int DEFAULT_QUEUE_PORT = 10001;
+
+    static final int DEFAULT_TABLE_PORT = 10002;
+
+    static final String DEFAULT_LOCATION = "/data";
+
+    /**
+     * The account name of the default credentials.
+     */
+    public static final String WELL_KNOWN_ACCOUNT_NAME = "devstoreaccount1";
+
+    /**
+     * The account key of the default credentials.
+     */
+    public static final String WELL_KNOWN_ACCOUNT_KEY =
+        "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==";
+
+    static final int NO_LIMIT = 0;
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse(
+        "mcr.microsoft.com/azure-storage/azurite"
+    );
+
+    private boolean useBlob = true;
+
+    private String blobHost = DEFAULT_HOST;
+
+    private int blobPort = DEFAULT_BLOB_PORT;
+
+    private boolean useQueue = true;
+
+    private String queueHost = DEFAULT_HOST;
+
+    private int queuePort = DEFAULT_QUEUE_PORT;
+
+    private boolean useTable = true;
+
+    private String tableHost = DEFAULT_HOST;
+
+    private int tablePort = DEFAULT_TABLE_PORT;
+
+    private String location = DEFAULT_LOCATION;
+
+    private boolean silent = false;
+
+    private String debug = null;
+
+    private boolean loose = false;
+
+    private boolean version = false;
+
+    private File cert = null;
+
+    private String certExtension = null;
+
+    private File key = null;
+
+    private String pwd = null;
+
+    private String oauth = null;
+
+    private boolean skipApiVersionCheck = false;
+
+    private boolean disableProductStyleUrl = false;
+
+    private boolean inMemoryPersistence = false;
+
+    private long extentMemoryLimit = NO_LIMIT;
+
+    private final List<String> azuriteAccounts = new ArrayList<>();
+
+    /**
+     * @param dockerImageName specified docker image name to run
+     */
+    public AzuriteContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+    }
+
+    /**
+     * Configure SSL with a custom certificate and password.
+     *
+     * @param pfxCert  The PFX certificate file
+     * @param password The password securing the certificate
+     * @return this
+     */
+    public AzuriteContainer withSsl(final File pfxCert, final String password) {
+        cert = pfxCert;
+        pwd = password;
+        certExtension = ".pfx";
+        return this;
+    }
+
+    /**
+     * Configure SSL with a custom certificate and private key.
+     *
+     * @param pemCert The PEM certificate file
+     * @param pemKey  The PEM key file
+     * @return this
+     */
+    public AzuriteContainer withSsl(final File pemCert, final File pemKey) {
+        cert = pemCert;
+        key = pemKey;
+        certExtension = ".pem";
+        return this;
+    }
```

Comment:
provide test

Reply:
Thanks, I will add some self-signed certs with 10 years of expiry as test resources to be used and implement some tests using them...

Reply:
Done, thank you! Went for 20 years certs covering localhost and 127.0.0.1 just in case.